addons/website_sale/views/templates.xml

```diff
@@ -2987,49 +3009,33 @@
+                </a>
             </div>
-            <div class="mb-4">
+            <div id="order_name" class="mb-4">
                 <h5>
-                    <em>
-                        <span>Order</span>
-                        <span t-field="order.name"/>
-                        <t t-if="order.state == 'sale'">
-                            <i class="fa fa-check-circle ms-1"/>
-                        </t>
-                    </em>
+                    <span>Order</span>
+                    <span t-field="order.name"/>
                 </h5>
             </div>
             <t t-if="request.env['res.users']._get_signup_invitation_scope() == 'b2c' and request.website.is_public_user()">
                 <p class="alert alert-info mt-3" role="status">
                     <a role="button" t-att-href="order.partner_id.signup_prepare() and order.partner_id.with_context(relative_url=True)._get_signup_url()" class="btn btn-primary o_translate_inline">Sign Up</a>
-                    to follow your order.
+                    <span class="align-middle">to follow your order.</span>
                 </p>
             </t>
             <div class="oe_structure clearfix mt-3" id="oe_structure_website_sale_confirmation_1"/>
-            <h4 class="text-start mt-3">Payment Information</h4>
-            <table class="table">
-                <tbody>
-                    <tr>
-                        <td colspan="2" class="ps-0">
-                            <t t-esc="tx_sudo.provider_id.name" />
-                        </td>
-                        <td class="text-end pe-0" width="100">
-                            <strong>Total:</strong>
-                        </td>
-                        <td class="text-end pe-0" width="100">
-                            <strong t-field="tx_sudo.amount" t-options="{'widget': 'monetary', 'display_currency': order.currency_id}" />
-                        </td>
-                    </tr>
-                </tbody>
-            </table>
             <t t-call="website_sale.payment_confirmation_status"/>
-            <div class="card mt-3">
-                <div class="card-body">
-                    <t t-set="same_shipping" t-value="bool(order.partner_shipping_id == order.partner_invoice_id or only_services)" />
-                    <div><b>Billing <t t-if="same_shipping and not only_services"> &amp; Delivery</t>: </b><span t-esc='order.partner_invoice_id' t-options="dict(widget='contact', fields=['address'], no_marker=True, separator=', ')" class="address-inline"/></div>
-                    <div t-if="not same_shipping and not only_services" groups="account.group_delivery_invoice_address"><b>Delivery: </b><span t-esc='order.partner_shipping_id' t-options="dict(widget='contact', fields=['address'], no_marker=True, separator=', ')"  class="address-inline"/></div>
-                </div>
+            <div class="mt-3">
+                <t t-call="website_sale.address_on_checkout"/>
```

Comment:
I'm not convinced at all by this change. I think it's not clear enough and with ten payments method or more you don't see the address at all. IMO the payment page should be reassuring, listing every details as it's the last page you get to see before confirming the order.

cc @Jean-boje

Reply:
I've discussed with boje and he wants to merge it like this, and adapt it later if needed